### PR TITLE
OpenZFSify CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

.github/CONTRIBUTING.md seems to have been hidden from greps for zfsonlinux when we moved to the new organization name.

### Description

Update stale references to "ZFS on Linux" to "OpenZFS" in CONTRIBUTING.md.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,9 @@
-# Contributing to ZFS on Linux
+# Contributing to OpenZFS
 <p align="center"><img src="http://zfsonlinux.org/images/zfs-linux.png"/></p>
```
Comment (freqlabs) [inline .github/CONTRIBUTING.md:2]: I can change this image to the one used in README.md

**Comment by freqlabs** [comment]
* Updated header imag to the OpenZFS image from README.md

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
-# Contributing to ZFS on Linux
-<p align="center"><img src="http://zfsonlinux.org/images/zfs-linux.png"/></p>
+# Contributing to OpenZFS
+<p align="center">
+![img](https://openzfs.github.io/openzfs-docs/_static/img/logo/480px-Open-ZFS-Secondary-Logo-Colour-halfsize.png)
```
Comment (behlendorf) [inline .github/CONTRIBUTING.md:3]: I'm not sure this syntax is quite right.  It doesn't appear to be correctly showing the new image in the rich diff output.

Comment (freqlabs) [reply .github/CONTRIBUTING.md]: I copied the line directly from README.md but I guess there is some nuance in mixing HTML and markdown I wasn't aware of. Sticking to HTML here works :)

**Pull request merged**